### PR TITLE
Code changes to output simulated GOES-18 ABI reflectances

### PR DESCRIPTION
**Pull request by EricJames-NOAA**
Code changes from SSEC / Wisconsin to output simulated satellite brightness temperatures from RRFS.  There are 10 new potential variables added to post_avblflds.xml, of which 9 are now included in fv3lam_rrfs.xml and the resulting postxconfig-NT.txt.  

The code changes were made by William Lewis, and have been tested for RRFS_NA_3km on Jet.  

There is a companion PR for regional_workflow, enabling use of new CRTMv2.4.0, for UPP only (retaining use of CRTMv2.3.0 for GSI and other applications).

### CODE REVIEW

File: modulefiles/jet [hunk 1]
```diff
@@ -22,7 +22,7 @@ module load hdf5/1.10.6
 module load netcdf/4.7.4
 
 module load bacio/2.4.1
-module load crtm/2.3.0
+module load crtm/2.4.0
```
Comment (hu5970) [inline modulefiles/jet]: This will make GSI use the 2.4.0.
Did you test GSI with this crtm/2.4.0. library?

File: parm/fv3lam_rrfs.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <postxml xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="EMC_POST_CTRL_Schema.xsd">
 <!--  Lin Gan:  20150427 validated -->
     <paramset>
-    <datset>PRSLEV</datset>
+    <datset>BGDAWP</datset>
```
Comment (hu5970) [inline parm/fv3lam_rrfs.xml]: PRSLEV is the same used for operation. Should not change this.
The script will make the file name change for generating BGDAWP for now. But we will get rid of this file name when downstream systems are ready.

File: parm/fv3lam_rrfs.xml [hunk 3]
```diff
@@ -1877,10 +1877,64 @@
...
+           <shortname>SBTA1816_ON_TOP_OF_ATMOS</shortname>
+           <pname>SBTA1816</pname>
+           <scale>4.0</scale>
+       </param>
+
     </paramset>
 
     <paramset>
-    <datset>NATLEV</datset>
+    <datset>BGRD3D</datset>
```
Comment (hu5970) [inline parm/fv3lam_rrfs.xml]: NATLEV is the name for operation. Can not be changed.

File: parm/post_avblflds.xml [hunk 4]
```diff
@@ -6432,23 +6512,23 @@
...
        <fixed_sfc1_type>entire_atmos</fixed_sfc1_type>
        <scale>3.0</scale>
     </param>
 
     <param>
        <post_avblfldidx>703</post_avblfldidx>
        <shortname>GSD_MAX_LTG_THREAT2_ON_ENTIRE_ATMOS</shortname>
-       <pname>LTGTHREAT2</pname>
+       <pname>LTNGTH2</pname>
        <fixed_sfc1_type>entire_atmos</fixed_sfc1_type>
```
Comment (hu5970) [inline parm/post_avblflds.xml]: Why change this one? It  is not related to this work.

File: parm/post_avblflds.xml [hunk 5]
```diff
@@ -6432,23 +6512,23 @@
     <param>
        <post_avblfldidx>702</post_avblfldidx>
        <shortname>GSD_MAX_LTG_THREAT1_ON_ENTIRE_ATMOS</shortname>
-       <pname>LTGTHREAT1</pname>
+       <pname>LTNGTH1</pname>
```
Comment (hu5970) [inline parm/post_avblflds.xml]: Please double check this one?

File: parm/post_avblflds.xml [hunk 6]
```diff
@@ -6432,23 +6512,23 @@
...
+       <pname>LTNGTH2</pname>
        <fixed_sfc1_type>entire_atmos</fixed_sfc1_type>
        <scale>3.0</scale>
     </param>
 
     <param>
        <post_avblfldidx>704</post_avblfldidx>
        <shortname>GSD_MAX_LTG_THREAT3_ON_ENTIRE_ATMOS</shortname>
-       <pname>LTNG</pname>
+       <pname>LTNGTH1</pname>
```
Comment (hu5970) [inline parm/post_avblflds.xml]: Please double check this one.

File: modulefiles/jet (hunk 1 above)
Comment (EricJames-NOAA) [reply modulefiles/jet]: No, I did not test GSI. Maybe it's OK to leave it as crtm/2.3.0 since UPP just uses the binary files, which are copied in from v2.4.0 by the updated scripts.

File: parm/fv3lam_rrfs.xml (hunk 2 above)
Comment (EricJames-NOAA) [reply parm/fv3lam_rrfs.xml]: OK.  Sorry, I will revert this.

File: parm/post_avblflds.xml (hunk 4 above)
Comment (EricJames-NOAA) [reply parm/post_avblflds.xml]: I will revert this.  SSEC / Wisconsin colleagues also want to have lightning threat output in RRFS, I guess they were going ahead and making changes in anticipation of that.

File: parm/postxconfig-NT-fv3lam_rrfs.txt [hunk 7]
```diff
@@ -7793,7 +7793,7 @@ UEID_ON_EFL
 1
 tmpl4_0
 UEID
-NCEP
+?
```
Comment (guoqing-noaa) [inline parm/postxconfig-NT-fv3lam_rrfs.txt]: I think we decide to use "NCEP" for this?

Comment (EricJames-NOAA) [reply parm/postxconfig-NT-fv3lam_rrfs.txt]: @guoqing-noaa I agree.  Do you have any hypothesis on where this change came from?  I don't think I made any change that could affect this.  Maybe the previous postxconfig-NT.txt was not consistent with the previous post_avblflds.xml and fv3lam_rrfs.xml

**Comment by EricJames-NOAA** [comment]
@hu5970 I heard back from Will Lewis on the new code in MISCLN.f...he says all those IGET numbers need to be there, otherwise the code segfaults later when the brightness temperature values are being output.  

I think the new static files are now OK, and they are consistent between this PR and the one for regional_workflow, with only the new fields added to postxconfig-NT-fv3lam_rrfs.txt.  I'm just doing a final test to check if the RRFS output is identical other than the new fields.

Thanks for your help!

**Comment by EricJames-NOAA** [comment]
@hu5970 sorry for the delay.  My tests revealed we still have differences in values in two solar irradiance variables, DSWRF and USWRF, in the native level files, but not in the pressure level files.  I confirmed that the issue is coming from the postxconfig-NT.txt file, since I was able to match those variables exactly using our new merged code and the old postxconfig-NT.txt file used currently for RRFS-B.  Let me know if you have any suggestions.  I asked Will Lewis to look at it too.

**Comment by EricJames-NOAA** [comment]
@hu5970 sorry for the delay.  My tests revealed we still have differences in values in two solar irradiance variables, DSWRF and USWRF, in the native level files, but not in the pressure level files.  I confirmed that the issue is coming from the postxconfig-NT.txt file, since I was able to match those variables exactly using our new merged code and the old postxconfig-NT.txt file used currently for RRFS-B.  Let me know if you have any suggestions.  I asked Will Lewis to look at it too.

**Comment by hu5970** [comment]
@EricJames-NOAA  It is strange that postxconfig-NT.txt can cause different to DSWRF and USWRF if the only difference of the postxconfig-NT.txt is new added GOES 18 ABI channels.

File: parm/params_grib2_tbl_new [hunk 8]
```diff
@@ -294,7 +294,6 @@
    0   7   204   1 EFHL
    0   3   222   1 EFSH
    0   7     9   0 EHLX
-   2   0   237   1 EIWATER
```
Comment (hu5970) [inline parm/params_grib2_tbl_new:297]: Does this intended to be removed?

The same are LAKEFRC and LANDFRAC.

There are new added code for GOES 18 ABI. But why to remove those definition that are not related to GOES ABI. A results from sync with RRFS fix?

File: parm/params_grib2_tbl_new [hunk 9]
```diff
@@ -1217,7 +1207,6 @@
    0   2   192   1 VWSH
   10   4    17   0 WATDENA
   10   4    16   0 WATERDEN
-   2   0   236   1 WATERSA
```
Comment (hu5970) [inline parm/params_grib2_tbl_new:1220]: parames_grib2_tbl_new.text and params_grib2_table_new have the same definitions of the variables but in different old. Do you know why those variables (like WATERSA) are removed?

File: parm/params_grib2_tbl_new (hunk 8 above)
Comment (EricJames-NOAA) [reply parm/params_grib2_tbl_new:297]: Yes, those removals are from syncing with RRFS fix.  As we discussed yesterday, it seems the UPP repository params_grib2_tbl_new file is updated to include more variables than in our realtime file, but we can deal with the differences when we sync with EMC.

**Comment by EricJames-NOAA** [comment]
@hu5970 When I output the new SBTA fields in the NATLEV file instead of PRSLEV file, there is no longer a problem with DSWRF or USWRF.  I get identical results to current realtime RRFS-B except for the new fields.   Is it OK to take this approach for now?  I have updated the PR.

**Review by hu5970 — APPROVED** [review]

**Comment by hu5970** [comment]
@EricJames-NOAA Please merge the UPP PR. We will check the PR for regional workflow next week.
